### PR TITLE
Fog probability

### PR DESCRIPTION
**Pull request by GlennGoddard**
Fixed part of it since I still had the wind speed in mph vice m/s.
Temperatures were already in C, so no change there.
I think something else is different, but this should be better and I will continue to compare.

I was working on the Snow_Probability already so I just commented it out for this PR and will continue with it.

### CODE REVIEW

**Comment by briis** [comment]
OK. I will look at it in a day, so just keep pushing to this PR.

**Comment by GlennGoddard** [comment]
@briis Just to verify since there have been some changes:
Refering to calls inside the integration and not what is returned to user
Is air_temperature always in C ?
Is dew_point always in C ?
Is freezing_level always in meters?
Is cloud_base always in meters?
Is wet_bulb always in meters?

I'm pretty sure station_height passes exactly what is put as an environmental variable in metric.

**Comment by briis** [comment]
Is air_temperature always in C ? Yes. This read directly from the Raw data
Is dew_point always in C ? Yes. This read directly from the Raw data
Is freezing_level always in meters? No. This might be feet or meters
Is cloud_base always in meters? No. This might be feet or meters
Is wet_bulb always in meters? No. This might be F or C

**Comment by GlennGoddard** [comment]
Snow Probability is done with the exception of passing if it is metric or imperial to the snow_probability function.  I ran out of time to find that for now.  I will see if I can find it later.

**Comment by GlennGoddard** [comment]
I just remembered I still need to update ReadMe also for snow_probability

**Comment by GlennGoddard** [comment]
@briis Please look at the "is_metric" of snow_probability.  That is the only thing I'm not sure I did right.
Other than that this PR should be good to go for both Fog and Snow Probability.

File: weatherflow2mqtt/helpers.py [hunk 1]
```diff
@@ -988,3 +993,52 @@ def fog_probability(self, solar_elevation, wind_speed, humidity, dew_point, air_
...
+            freezing_level = freezing_level / 3.28
+            cloud_base = cloud_base / 3.28
+            dew_point = dew_point
+            wet_bulb = (wet_bulb - 32) / 1.8
+            station_height = station_height
+        
+        snow_line = freezing_level - 228.6 # 750 ft / 228.6 m, snow line can vary in distance from freezing line
+        dptt = dew_point + temperature
+        
+        if ((air_temperature <= 2.1) && (snow_line <= station_height) && (dew_point <= 1) && (wet_bulb <= 2.1) && (freezing_line <= cloud_base) && (air_temperature >= -40)):
```
Comment (briis) [inline weatherflow2mqtt/helpers.py:1037]: You use a variable `freezing_line`, which is not supplied to the formula, we only have `freezing_level`. Should it be changed or where should the value come from?

Don't make any changes in this PR, as I have already merged it and fixed some minor issues, plus simplified the is_metric routine. 
`&&` in Python is `and` 😁

Comment (briis) [reply weatherflow2mqtt/helpers.py:1037]: I now assumed that `freezing_line` is the same as `freezing_level`, so I used that value in the formula.

**Comment by briis** [comment]
A new `dev` build is now uploaded to Docker Hub, which contains the latest modifications and additions

**Comment by GlennGoddard** [comment]
Yes, freezing_line is freezing_level.  I'm glad you posted where I used it (good catch); I'll do a new PR to update that.  There is something still off on Fog Probability that I need to do more troubleshooting on.

Yep, I missed changing the '&&' to 'and'.  Coding and sleep deprivation sometimes work and sometimes don't.

I'll fix it in my current conditions branch.

**Pull request merged**